src/runtime_src/core/pcie/windows/device_windows.cpp

```diff
@@ -198,6 +198,152 @@ info_mgmt(const device_type* device, qr_type qr, const std::type_info&, boost::a
   }
 }
 
+static void
+sensor_info(const device_type* device, qr_type qr, const std::type_info&, boost::any& value)
+{
+  auto init_sensor_info = [](const device_type* dev) {
+    xcl_sensor info = { 0 };
+    userpf::get_sensor_info(dev->get_user_handle(), &info);
+    return info;
+  };
+
+  static std::map<const device_type*, xcl_sensor> info_map;
+  static std::mutex mutex;
+  std::lock_guard<std::mutex> lk(mutex);
+  auto it = info_map.find(device);
+  if (it == info_map.end()) {
+    auto ret = info_map.emplace(device,init_sensor_info(device));
```

Comment:
You are caching the sensor info, when does it expire?  Isn't sensor info dynamic data?  It's fine to cache, but maybe expire every second or so?

Reply:
Defer till we actually need sensor  information in a long running process like xbutil top.